// @ngx-formly/bootstrap/esm2015/radio/radio.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlySelectModule } from '@ngx-formly/core/select';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';
import { FormlyFieldRadio } from './radio.type';
let FormlyBootstrapRadioModule = class FormlyBootstrapRadioModule {
};
FormlyBootstrapRadioModule = __decorate([
    NgModule({
        declarations: [FormlyFieldRadio],
        imports: [
            CommonModule,
            ReactiveFormsModule,
            FormlyBootstrapFormFieldModule,
            FormlySelectModule,
            FormlyModule.forChild({
                types: [
                    {
                        name: 'radio',
                        component: FormlyFieldRadio,
                        wrappers: ['form-field'],
                    },
                ],
            }),
        ],
    })
], FormlyBootstrapRadioModule);
export { FormlyBootstrapRadioModule };
