// @ngx-formly/bootstrap/esm5/form-field/form-field.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyWrapperFormField } from './form-field.wrapper';
var FormlyBootstrapFormFieldModule = /** @class */ (function () {
    function FormlyBootstrapFormFieldModule() {
    }
    FormlyBootstrapFormFieldModule = __decorate([
        NgModule({
            declarations: [FormlyWrapperFormField],
            imports: [
                CommonModule,
                ReactiveFormsModule,
                FormlyModule.forChild({
                    wrappers: [
                        {
                            name: 'form-field',
                            component: FormlyWrapperFormField,
                        },
                    ],
                }),
            ],
        })
    ], FormlyBootstrapFormFieldModule);
    return FormlyBootstrapFormFieldModule;
}());
export { FormlyBootstrapFormFieldModule };
