// @ngx-formly/bootstrap/esm5/select/select.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyModule } from '@ngx-formly/core';
import { FormlySelectModule } from '@ngx-formly/core/select';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';
import { FormlyFieldSelect } from './select.type';
var FormlyBootstrapSelectModule = /** @class */ (function () {
    function FormlyBootstrapSelectModule() {
    }
    FormlyBootstrapSelectModule = __decorate([
        NgModule({
            declarations: [FormlyFieldSelect],
            imports: [
                CommonModule,
                ReactiveFormsModule,
                FormlyBootstrapFormFieldModule,
                FormlySelectModule,
                FormlyModule.forChild({
                    types: [
                        {
                            name: 'select',
                            component: FormlyFieldSelect,
                            wrappers: ['form-field'],
                        },
                        { name: 'enum', extends: 'select' },
                    ],
                }),
            ],
        })
    ], FormlyBootstrapSelectModule);
    return FormlyBootstrapSelectModule;
}());
export { FormlyBootstrapSelectModule };
