// @ngx-formly/bootstrap/esm5/multicheckbox/multicheckbox.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyModule } from '@ngx-formly/core';
import { FormlySelectModule } from '@ngx-formly/core/select';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';
import { FormlyFieldMultiCheckbox } from './multicheckbox.type';
var FormlyBootstrapMultiCheckboxModule = /** @class */ (function () {
    function FormlyBootstrapMultiCheckboxModule() {
    }
    FormlyBootstrapMultiCheckboxModule = __decorate([
        NgModule({
            declarations: [FormlyFieldMultiCheckbox],
            imports: [
                CommonModule,
                ReactiveFormsModule,
                FormlyBootstrapFormFieldModule,
                FormlySelectModule,
                FormlyModule.forChild({
                    types: [
                        {
                            name: 'multicheckbox',
                            component: FormlyFieldMultiCheckbox,
                            wrappers: ['form-field'],
                        },
                    ],
                }),
            ],
        })
    ], FormlyBootstrapMultiCheckboxModule);
    return FormlyBootstrapMultiCheckboxModule;
}());
export { FormlyBootstrapMultiCheckboxModule };
