// @ngx-formly/bootstrap/esm5/textarea/textarea.type.js
import { __decorate, __extends } from "tslib";
import { Component, ChangeDetectionStrategy } from '@angular/core';
import { FieldType } from '@ngx-formly/core';
var FormlyFieldTextArea = /** @class */ (function (_super) {
    __extends(FormlyFieldTextArea, _super);
    function FormlyFieldTextArea() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.defaultOptions = {
            templateOptions: {
                cols: 1,
                rows: 1,
            },
        };
        return _this;
    }
    FormlyFieldTextArea = __decorate([
        Component({
            selector: 'formly-field-textarea',
            template: "\n    <textarea\n      [formControl]=\"formControl\"\n      [cols]=\"to.cols\"\n      [rows]=\"to.rows\"\n      class=\"form-control\"\n      [class.is-invalid]=\"showError\"\n      [formlyAttributes]=\"field\"\n    >\n    </textarea>\n  ",
            changeDetection: ChangeDetectionStrategy.OnPush
        })
    ], FormlyFieldTextArea);
    return FormlyFieldTextArea;
}(FieldType));
export { FormlyFieldTextArea };
