// @ngx-formly/bootstrap/esm5/input/input.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';
import { FormlyFieldInput } from './input.type';
var FormlyBootstrapInputModule = /** @class */ (function () {
    function FormlyBootstrapInputModule() {
    }
    FormlyBootstrapInputModule = __decorate([
        NgModule({
            declarations: [FormlyFieldInput],
            imports: [
                CommonModule,
                ReactiveFormsModule,
                FormlyBootstrapFormFieldModule,
                FormlyModule.forChild({
                    types: [
                        {
                            name: 'input',
                            component: FormlyFieldInput,
                            wrappers: ['form-field'],
                        },
                        { name: 'string', extends: 'input' },
                        {
                            name: 'number',
                            extends: 'input',
                            defaultOptions: {
                                templateOptions: {
                                    type: 'number',
                                },
                            },
                        },
                        {
                            name: 'integer',
                            extends: 'input',
                            defaultOptions: {
                                templateOptions: {
                                    type: 'number',
                                },
                            },
                        },
                    ],
                }),
            ],
        })
    ], FormlyBootstrapInputModule);
    return FormlyBootstrapInputModule;
}());
export { FormlyBootstrapInputModule };
