// @ngx-formly/bootstrap/esm2015/textarea/textarea.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';
import { FormlyFieldTextArea } from './textarea.type';
let FormlyBootstrapTextAreaModule = class FormlyBootstrapTextAreaModule {
};
FormlyBootstrapTextAreaModule = __decorate([
    NgModule({
        declarations: [FormlyFieldTextArea],
        imports: [
            CommonModule,
            ReactiveFormsModule,
            FormlyBootstrapFormFieldModule,
            FormlyModule.forChild({
                types: [
                    {
                        name: 'textarea',
                        component: FormlyFieldTextArea,
                        wrappers: ['form-field'],
                    },
                ],
            }),
        ],
    })
], FormlyBootstrapTextAreaModule);
export { FormlyBootstrapTextAreaModule };
