// @ngx-formly/bootstrap/esm5/checkbox/checkbox.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';
import { FormlyFieldCheckbox } from './checkbox.type';
var FormlyBootstrapCheckboxModule = /** @class */ (function () {
    function FormlyBootstrapCheckboxModule() {
    }
    FormlyBootstrapCheckboxModule = __decorate([
        NgModule({
            declarations: [FormlyFieldCheckbox],
            imports: [
                CommonModule,
                ReactiveFormsModule,
                FormlyBootstrapFormFieldModule,
                FormlyModule.forChild({
                    types: [
                        {
                            name: 'checkbox',
                            component: FormlyFieldCheckbox,
                            wrappers: ['form-field'],
                        },
                        {
                            name: 'boolean',
                            extends: 'checkbox',
                        },
                    ],
                }),
            ],
        })
    ], FormlyBootstrapCheckboxModule);
    return FormlyBootstrapCheckboxModule;
}());
export { FormlyBootstrapCheckboxModule };
