// @ngx-formly/bootstrap/esm5/checkbox/checkbox.type.js
import { __decorate, __extends } from "tslib";
import { Component, ChangeDetectionStrategy } from '@angular/core';
import { FieldType } from '@ngx-formly/core';
var FormlyFieldCheckbox = /** @class */ (function (_super) {
    __extends(FormlyFieldCheckbox, _super);
    function FormlyFieldCheckbox() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.defaultOptions = {
            templateOptions: {
                indeterminate: true,
                hideLabel: true,
                formCheck: 'custom',
            },
        };
        return _this;
    }
    FormlyFieldCheckbox = __decorate([
        Component({
            selector: 'formly-field-checkbox',
            template: "\n    <div\n      [ngClass]=\"{\n        'form-check': to.formCheck.indexOf('custom') === -1,\n        'form-check-inline': to.formCheck === 'inline',\n        'custom-control': to.formCheck.indexOf('custom') === 0,\n        'custom-checkbox': to.formCheck === 'custom' || to.formCheck === 'custom-inline',\n        'custom-control-inline': to.formCheck === 'custom-inline',\n        'custom-switch': to.formCheck === 'custom-switch'\n      }\"\n    >\n      <input\n        type=\"checkbox\"\n        [class.is-invalid]=\"showError\"\n        [class.form-check-input]=\"to.formCheck.indexOf('custom') === -1\"\n        [class.position-static]=\"to.formCheck === 'nolabel'\"\n        [class.custom-control-input]=\"to.formCheck.indexOf('custom') === 0\"\n        [indeterminate]=\"to.indeterminate && formControl.value == null\"\n        [formControl]=\"formControl\"\n        [formlyAttributes]=\"field\"\n      />\n      <label\n        *ngIf=\"to.formCheck !== 'nolabel'\"\n        [for]=\"id\"\n        [class.form-check-label]=\"to.formCheck.indexOf('custom') === -1\"\n        [class.custom-control-label]=\"to.formCheck.indexOf('custom') === 0\"\n      >\n        {{ to.label }}\n        <span *ngIf=\"to.required && to.hideRequiredMarker !== true\">*</span>\n      </label>\n    </div>\n  ",
            changeDetection: ChangeDetectionStrategy.OnPush
        })
    ], FormlyFieldCheckbox);
    return FormlyFieldCheckbox;
}(FieldType));
export { FormlyFieldCheckbox };
