// @ngx-formly/bootstrap/fesm5/ngx-formly-bootstrap-form-field.js
import { __extends, __decorate } from 'tslib';
import { Component, ChangeDetectionStrategy, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldWrapper, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';

var FormlyWrapperFormField = /** @class */ (function (_super) {
    __extends(FormlyWrapperFormField, _super);
    function FormlyWrapperFormField() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FormlyWrapperFormField = __decorate([
        Component({
            selector: 'formly-wrapper-form-field',
            template: "\n    <div class=\"form-group\" [class.has-error]=\"showError\">\n      <label *ngIf=\"to.label && to.hideLabel !== true\" [attr.for]=\"id\">\n        {{ to.label }}\n        <span *ngIf=\"to.required && to.hideRequiredMarker !== true\">*</span>\n      </label>\n\n      <ng-template #fieldComponent></ng-template>\n\n      <div *ngIf=\"showError\" class=\"invalid-feedback\" [style.display]=\"'block'\">\n        <formly-validation-message [field]=\"field\"></formly-validation-message>\n      </div>\n\n      <small *ngIf=\"to.description\" class=\"form-text text-muted\">{{ to.description }}</small>\n    </div>\n  ",
            changeDetection: ChangeDetectionStrategy.OnPush
        })
    ], FormlyWrapperFormField);
    return FormlyWrapperFormField;
}(FieldWrapper));

var FormlyBootstrapFormFieldModule = /** @class */ (function () {
    function FormlyBootstrapFormFieldModule() {
    }
    FormlyBootstrapFormFieldModule = __decorate([
        NgModule({
            declarations: [FormlyWrapperFormField],
            imports: [
                CommonModule,
                ReactiveFormsModule,
                FormlyModule.forChild({
                    wrappers: [
                        {
                            name: 'form-field',
                            component: FormlyWrapperFormField,
                        },
                    ],
                }),
            ],
        })
    ], FormlyBootstrapFormFieldModule);
    return FormlyBootstrapFormFieldModule;
}());

/**
 * Generated bundle index. Do not edit.
 */

export { FormlyBootstrapFormFieldModule, FormlyWrapperFormField as ɵa };
