// @ngx-formly/bootstrap/fesm5/ngx-formly-bootstrap-textarea.js
import { __extends, __decorate } from 'tslib';
import { Component, ChangeDetectionStrategy, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldType, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';

var FormlyFieldTextArea = /** @class */ (function (_super) {
    __extends(FormlyFieldTextArea, _super);
    function FormlyFieldTextArea() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.defaultOptions = {
            templateOptions: {
                cols: 1,
                rows: 1,
            },
        };
        return _this;
    }
    FormlyFieldTextArea = __decorate([
        Component({
            selector: 'formly-field-textarea',
            template: "\n    <textarea\n      [formControl]=\"formControl\"\n      [cols]=\"to.cols\"\n      [rows]=\"to.rows\"\n      class=\"form-control\"\n      [class.is-invalid]=\"showError\"\n      [formlyAttributes]=\"field\"\n    >\n    </textarea>\n  ",
            changeDetection: ChangeDetectionStrategy.OnPush
        })
    ], FormlyFieldTextArea);
    return FormlyFieldTextArea;
}(FieldType));

var FormlyBootstrapTextAreaModule = /** @class */ (function () {
    function FormlyBootstrapTextAreaModule() {
    }
    FormlyBootstrapTextAreaModule = __decorate([
        NgModule({
            declarations: [FormlyFieldTextArea],
            imports: [
                CommonModule,
                ReactiveFormsModule,
                FormlyBootstrapFormFieldModule,
                FormlyModule.forChild({
                    types: [
                        {
                            name: 'textarea',
                            component: FormlyFieldTextArea,
                            wrappers: ['form-field'],
                        },
                    ],
                }),
            ],
        })
    ], FormlyBootstrapTextAreaModule);
    return FormlyBootstrapTextAreaModule;
}());

/**
 * Generated bundle index. Do not edit.
 */

export { FormlyBootstrapTextAreaModule, FormlyFieldTextArea };
