// @ngx-formly/bootstrap/fesm2015/ngx-formly-bootstrap-textarea.js
import { __decorate } from 'tslib';
import { Component, ChangeDetectionStrategy, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FieldType, FormlyModule } from '@ngx-formly/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyBootstrapFormFieldModule } from '@ngx-formly/bootstrap/form-field';

let FormlyFieldTextArea = class FormlyFieldTextArea extends FieldType {
    constructor() {
        super(...arguments);
        this.defaultOptions = {
            templateOptions: {
                cols: 1,
                rows: 1,
            },
        };
    }
};
FormlyFieldTextArea = __decorate([
    Component({
        selector: 'formly-field-textarea',
        template: `
    <textarea
      [formControl]="formControl"
      [cols]="to.cols"
      [rows]="to.rows"
      class="form-control"
      [class.is-invalid]="showError"
      [formlyAttributes]="field"
    >
    </textarea>
  `,
        changeDetection: ChangeDetectionStrategy.OnPush
    })
], FormlyFieldTextArea);

let FormlyBootstrapTextAreaModule = class FormlyBootstrapTextAreaModule {
};
FormlyBootstrapTextAreaModule = __decorate([
    NgModule({
        declarations: [FormlyFieldTextArea],
        imports: [
            CommonModule,
            ReactiveFormsModule,
            FormlyBootstrapFormFieldModule,
            FormlyModule.forChild({
                types: [
                    {
                        name: 'textarea',
                        component: FormlyFieldTextArea,
                        wrappers: ['form-field'],
                    },
                ],
            }),
        ],
    })
], FormlyBootstrapTextAreaModule);

/**
 * Generated bundle index. Do not edit.
 */

export { FormlyBootstrapTextAreaModule, FormlyFieldTextArea };
